e2e/run/run.go

```diff
@@ -295,19 +295,126 @@ func (c ctx) testFuseExt3Mount(t *testing.T) {
 	)
 }
 
+func (c ctx) testAddPackageWithFakerootAndTmpfs(t *testing.T) {
+	tempDir, cleanup := e2e.MakeTempDir(t, c.env.TestDir, "", "")
+	defer e2e.Privileged(cleanup)
+
+	sandbox, err := ioutil.TempDir(tempDir, "sandbox")
+	if err != nil {
+		t.Fatalf("could not create sandbox folder inside tempdir: %s", tempDir)
+	}
+
+	sif := fmt.Sprintf("%s/centos7.sif", tempDir)
+
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs(sif, "docker://centos:7"),
+		e2e.ExpectExit(0),
+	)
+
+	c.env.RunApptainer(t,
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("--sandbox", "--force", sandbox, sif),
+		e2e.ExpectExit(0),
+	)
+
+	// we need to increase sessiondir max size to 32MB
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.RootProfile),
+		e2e.WithCommand("config"),
+		e2e.WithArgs("global", "-s", "sessiondir max size", "1024"),
+		e2e.ExpectExit(0),
+	)
+
+	// restore sessiondir max size to 16MB
+	defer c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.RootProfile),
+		e2e.WithCommand("config"),
+		e2e.WithArgs("global", "-s", "sessiondir max size", "16"),
+		e2e.ExpectExit(0),
+	)
+
+	// running under the mode 1, 1a (--with-suid) (https://apptainer.org/docs/user/main/fakeroot.html)
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.FakerootProfile),
+		e2e.WithCommand("exec"),
+		e2e.WithArgs("--writable-tmpfs", sandbox, "yum", "install", "-y", "openssh"),
+		e2e.ExpectExit(0), // works fine, sif will fail with not enough permission
```

Comment:
Ok I have done more testing with a lot of combinations and have indeed found that using a sif file with mode 1a does not work well, I'm not sure why.  Just please change the comment from `sif will fail ...` to `sif is not allowed` because I plan to put in an abort message telling people to try either a sandbox or `--userns` (`--ignore-subuid` also works but I don't think I'll mention that option).

Reply:
I take it back.  The problem is only happening when the overlay directory is a tmpfs filesystem, as I noted in issue #651.  I will do more investigation, maybe I can do something about that.  Worst case I'd rather have you create an image file with `apptainer overlay create --fakeroot --size 512 tempDir/overlay.img` and then use `--overlay tempDir/overlay.img` instead of `--writable-tmpfs`.  That works.  So does just an overlay dir on the filesystem.

Reply:
@DrDaveD 
Thanks for more tests and suggestions.
Refactor the code and squash the commits.